pkg/controller/cloudlet_resource_check.go

```diff
@@ -381,3 +382,43 @@ func (s *CloudletResCalc) getResourceLimits() ResLimitMap {
 	}
 	return limits
 }
+
+// calcResourceScore gets a score which represents the available resources
+// on a cloudlet. A higher score means more available resources.
+func (s *CloudletResCalc) calcResourceScore(usedVals resspec.ResValMap) uint64 {
+	// get max value for each resource on cloudlet
+	maxVals := getMaxResourceVals(s.deps.cloudletInfo.ResourcesSnapshot.Info, s.deps.cloudlet.ResourceQuotas)
+	// Calculate score based on weights and free values
+	// Because some resources may have no limit, track the number
+	// of resources we've scored. We'll divide by this number to
+	// get an average per-resource score for comparisons.
+	var score, numScored uint64
+	for res, weight := range resourceWeights {
```

Comment:
`resourceWeights` is in `potential_cloudlet.go`, but not even used there anymore. I suggest moving it to `cloudcommon` pkg, since map values are from that package anyways.

Reply:
Hm, I can move it to cloudlet_resource_check.go. I want to keep all the decision-making algorithm code inside the controller (or some specific decision alg package), as none of the other services should be making decisions about deployment placement.

Reply:
Ok, I see. In that case it doesn't really matter where that variable is declared - can just stay in potential_cloudlet.go